src/plugins/kibana_usage_collection/server/collectors/application_usage/schema.ts

```diff
@@ -48,8 +48,7 @@ const commonSchema: MakeSchemaFrom<ApplicationUsageTelemetryReport[string]> = {
   },
 };
 
-// These keys obtained by searching for `/application\w*\.register\(/` and checking the value of the attr `id`.
-// TODO: Find a way to update these keys automatically.
+// There is a test in x-pack/test/usage_collection that validates that the keys in here match all the registered apps
```

Comment:
As adding a new app will cause the test to fail, maybe there should be a comment on the test pointing to this file to help developers fix it when they'll add new apps?

Reply:
@pgayvallet makes sense! I've updated the test to point at the file in the error message.

Here's an example of the output:

```
0 passing (10.4s)
     │1 failing
     │
     │1)    Application Usage
     │       keys in the schema match the registered application IDs:
     │
     │
     │       Error: Application Usage's schema is not up-to-date with the actual registered apps. Please update it at src/plugins/kibana_usage_collection/server/collectors/application_usage/schema.ts.
     │ expected [ 'apm',
     │   'appSearch',
     │   'canvas',
     │   'dashboard_mode',
     │   'dashboards',
     │   'dev_tools',
     │   'discover',
     │   'enterpriseSearch',
     │   'error',
     │   'fleet',
     │   'graph',
     │   'home',
     │   'infra',
     │   'ingestManager',
     │   'kibana',
     │   'kibanaOverview',
     │   'lens',
     │   'logs',
     │   'management',
     │   'maps',
     │   'metrics',
     │   'ml',
     │   'monitoring',
     │   'observability-overview',
     │   'securitySolution',
     │   'securitySolution:administration',
     │   'securitySolution:case',
     │   'securitySolution:detections',
     │   'securitySolution:hosts',
     │   'securitySolution:network',
     │   'securitySolution:overview',
     │   'securitySolution:timelines',
     │   'security_access_agreement',
     │   'security_account',
     │   'security_capture_url',
     │   'security_logged_out',
     │   'security_login',
     │   'security_logout',
     │   'security_overwritten_session',
     │   'short_url_redirect',
     │   'siem',
     │   'space_selector',
     │   'status',
     │   'test',
     │   'timelion',
     │   'uptime',
     │   'ux',
     │   'visualize',
     │   'workplaceSearch' ] to sort of equal [ 'apm',
     │   'appSearch',
     │   'canvas',
     │   'dashboard_mode',
     │   'dashboards',
     │   'dev_tools',
     │   'discover',
     │   'enterpriseSearch',
     │   'error',
     │   'fleet',
     │   'graph',
     │   'home',
     │   'infra',
     │   'ingestManager',
     │   'kibana',
     │   'kibanaOverview',
     │   'lens',
     │   'logs',
     │   'management',
     │   'maps',
     │   'metrics',
     │   'ml',
     │   'monitoring',
     │   'observability-overview',
     │   'securitySolution',
     │   'securitySolution:administration',
     │   'securitySolution:case',
     │   'securitySolution:detections',
     │   'securitySolution:hosts',
     │   'securitySolution:network',
     │   'securitySolution:overview',
     │   'securitySolution:timelines',
     │   'security_access_agreement',
     │   'security_account',
     │   'security_capture_url',
     │   'security_logged_out',
     │   'security_login',
     │   'security_logout',
     │   'security_overwritten_session',
     │   'short_url_redirect',
     │   'siem',
     │   'space_selector',
     │   'status',
     │   'timelion',
     │   'uptime',
     │   'ux',
     │   'visualize',
     │   'workplaceSearch' ]
     │       + expected - actual
     │
     │          "short_url_redirect"
     │          "siem"
     │          "space_selector"
     │          "status"
     │       -  "test"
     │          "timelion"
     │          "uptime"
     │          "ux"
     │          "visualize"
     │
     │       at Assertion.assert (/Users/afharo/Developer/elastic/kibana/packages/kbn-expect/expect.js:100:11)
     │       at Assertion.eql (/Users/afharo/Developer/elastic/kibana/packages/kbn-expect/expect.js:244:8)
     │       at Context.<anonymous> (test/usage_collection/test_suites/application_usage/index.ts:21:63)
     │       at runMicrotasks (<anonymous>)
     │       at processTicksAndRejections (internal/process/task_queues.js:93:5)
     │       at Object.apply (/Users/afharo/Developer/elastic/kibana/packages/kbn-test/src/functional_test_runner/lib/mocha/wrap_function.js:84:16)
     │
     │
```

Reply:
Perfect! thanks.